mycodo/controllers/controller_pid.py

```diff
@@ -314,10 +350,8 @@ def setup_method(self, method_id):
         """ Initialize method variables to start running a method """
         self.setpoint_tracking_id = ''
 
-        method = db_retrieve_table_daemon(Method, unique_id=method_id)
-        method_data = db_retrieve_table_daemon(MethodData)
-        method_data = method_data.filter(MethodData.method_id == method_id)
-        method_data_repeat = method_data.filter(MethodData.duration_sec == 0).first()
+        method = load_method(method_id, self.logger)
+
```

Comment:
Previous implementation read database for method tables within the controller. This was migrated into the method code.